packages/app/src/cli/models/app/app.ts

```diff
@@ -202,6 +210,12 @@ export function appIsLaunchable(app: AppInterface) {
   return Boolean(frontendConfig || backendConfig)
 }
 
+export function filterNonVersionedAppFields(app: AppInterface) {
+  return Object.keys(app.configuration).filter(
+    (fieldName) => !Object.keys(NonVersionedAppSchema.shape).concat('path').includes(fieldName),
+  )
+}
```

Comment:
This should do it: `return Object.keys(VersionedAppSchema.parse(app.configuration))`

Reply:
I'm trying to get the fields in the current configuration file that are versioned.

`path`  is added to `app.configuration` content manually in other parts of the codebase (not included in the file) and we don't want to display it in the list of modified fields

> This should do it: return Object.keys(VersionedAppSchema.parse(app.configuration))

As mentioned in a previous comment, there's a following  [Generate app schema dynamically](https://github.com/Shopify/cli/pull/3191) PR so the fixed `VersionedAppSchema` is going to be removed because the schema will be generated dynamically from the `config` extension specs schemas